src/assets/TokenListController.ts

```diff
@@ -126,29 +135,43 @@ export class TokenListController extends BaseController<
     this.chainId = chainId;
     this.updatePreventPollingOnNetworkRestart(preventPollingOnNetworkRestart);
     this.abortController = new AbortController();
-    this.messagingSystem.subscribe(
-      'NetworkController:providerChange',
-      async (providerConfig) => {
-        if (this.chainId !== providerConfig.chainId) {
-          this.abortController.abort();
-          this.abortController = new AbortController();
-          this.chainId = providerConfig.chainId;
-          if (this.state.preventPollingOnNetworkRestart) {
-            this.clearingTokenListData();
-          } else {
-            // Ensure tokenList is referencing data from correct network
-            this.update(() => {
-              return {
-                ...this.state,
-                tokenList:
-                  this.state.tokensChainsCache[this.chainId]?.data || {},
-              };
-            });
-            await this.restart();
-          }
-        }
-      },
-    );
+    if (onNetworkStateChange) {
```

Comment:
Could you add a test for this case as well? Hopefully it won't be too difficult to resurrect something that was adjusted in the PR where this change was made

Reply:
done [here](https://github.com/MetaMask/controllers/pull/932/commits/8d084e1873e112db7bb7ed3233aa2538389bb7fd)

should probably do one for GasFeeController too eh?

Reply:
Yep, certainly!